tests/js/session-tests.js

```diff
@@ -742,13 +742,14 @@ module.exports = {
                 const config = { sync: { user, url: 'realm://localhost:9080/~/myrealm' } };
                 config.sync.error = (sender, error) => {
                     try {
+                        TestCase.assertEqual(error.name, 'ClientReset');
                         TestCase.assertEqual(error.code, 7); // 7 -> client reset
```

Comment:
do we need to `error.code` if we have the `error.name` here.

Reply:
If we should get rid of `error.code`, we need to map all error codes to names.

Reply:
I thought we introduced this one only here. I see your point.I guess if we decide to leave it then this error code should be coming from the source. I mean isn't it dangerous to assign some value here ourselves. since it may collide with others coming from the sync session handling code.

Reply:
Ok now I checked what `is_client_reset_requested` does here
https://github.com/realm/realm-db/blob/b0dd1381d89958474cce5584eee56d78e530a3e2/src/realm/object-store/sync/sync_config.hpp#L88

it seems it is better to leave the code as it is and client code to always compare with the name as `IncompatibleSyncedRealmError` does

I am seeing some value of having the original code present since it might help in certain scenarios. I mean to give the user these values might be of benefit to him (debugging is one of them)
``` 
ProtocolError::bad_server_file_ident
ProtocolError::bad_client_file_ident
ProtocolError::bad_server_version
ProtocolError::diverging_histories 
```

Reply:
I think the current functionality is closer to our other products: https://deploy-preview-4096--realm.netlify.com/docs/java/4.1.1/api/. This will make it easier for developers.